src/ComponentRegister.test.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+
+import {
+  ComponentRegister,
+  LinkComponent,
+  LinkComponentProps
+} from "./ComponentRegister";
+
+const TestLinkComponent: LinkComponent = ({
+  href,
+  children
+}: LinkComponentProps) => <div data-href={href}>{children}</div>;
+
+describe(".init()", () => {
+  it("registers all of the components provided", () => {
+    ComponentRegister.init({
+      components: {
+        Link: TestLinkComponent
+      }
+    });
+
+    expect(ComponentRegister.Link).toStrictEqual(TestLinkComponent);
+  });
+  it("does not alter properties if none are provided", () => {
+    ComponentRegister.init({});
+
+    expect(ComponentRegister.Link).toStrictEqual("a");
+  });
+});
+it("registers link by default as an anchor tag", () => {
```

Comment:
Let's wrap this in a `describe(".Link", () => {})` and rename the test to `defaults to "a"`, or similar. The test doesn't need to instantiate a component to test this. You should be able to do `expect(ComponentRegister.Link).toEqual("a")`.